Automattic/woocommerce-payments
# Consolidate gateways implementation - step II

This PR is the second step of https://github.com/Automattic/woocommerce-payments/issues/7782. The first part is covered by https://github.com/Automattic/woocommerce-payments/pull/7937.

#### Changes proposed in this Pull Request
https://github.com/Automattic/woocommerce-payments/pull/7937 merged the split UPE into the UPE gateway, leaving the merging of the UPE gateway into the gateway for this PR. Th main goal of this PR is to have the single gateway class for the WooPayments plugin. 

While merging, some unused areas of code (e.g. non-used variables) were removed from the codebase as well. 

> 💡 Unit tests for gateways merging as well as E2E tests merging are not in the scope of this PR. This is in order for us to be able to review this PR faster to possibly include it into the `7.0.0` release, leaving the (unrelated to production code) test improvements for the separate PR which I'll take care of.













#### Testing instructions





* enable multiple payment methods and confirm every gateway is displayed as expected on blocks & shortcode checkouts
* check some critical flows
    * 3DS payments (success & rejection)
    * [pay for a declined order](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#pay-for-order)
    * [changing payment method for subscriptions](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#change-subscription-payment-method-to-a-new-card)
* confirm that `save payment method` checkbox is displayed both on shortcode and blocks checkout
* enable WooPay and confirm that
    * payments via WooPay checkout are working as expected (additionally for this use-case please confirm that the WooPay's `save payment method` checkbox is displayed)
    * payments via shortcode checkout are working correctly (check CC & one non-card payment method)
    * payments via blocks checkout are working correctly (check CC & one non-card payment method)
* after performing successful payments, check in the WooCommerce -> `Orders` if the order is displayed correctly
* check the successful payment's details on Stripe by confirming that the status is correct and potentially performing a high-level sanity check of the request/metadata fields
* disable WooPay, enable Stripe Link, perform a payment with Stripe Link and confirm there are no unexpected errors

This PR touches main classes so it's hard to pin-point some explicit test cases. I expect plugin to either fail at some major stages (like e.g. opening the checkout page) or work correctly everywhere. Besides the above list of testing instructions, feel free to additionally check anything you might find worth checking after going through the code. 

-------------------


**Post merge**
